src/common/messages.rs

```diff
@@ -35,90 +46,103 @@ pub fn format_get_external_ip_message() -> String {
 }
 
 pub fn format_add_any_port_mapping_message(
+    schema: &Vec<String>,
     protocol: PortMappingProtocol,
     external_port: u16,
     local_addr: SocketAddrV4,
     lease_duration: u32,
     description: &str,
 ) -> String {
-    format!("<?xml version=\"1.0\"?>
-<s:Envelope xmlns:s=\"http://schemas.xmlsoap.org/soap/envelope/\" s:encodingStyle=\"http://schemas.xmlsoap.org/soap/encoding/\">
-<s:Body>
-    <u:AddAnyPortMapping xmlns:u=\"urn:schemas-upnp-org:service:WANIPConnection:1\">
-        <NewProtocol>{}</NewProtocol>
-        <NewExternalPort>{}</NewExternalPort>
-        <NewInternalClient>{}</NewInternalClient>
-        <NewInternalPort>{}</NewInternalPort>
-        <NewLeaseDuration>{}</NewLeaseDuration>
-        <NewPortMappingDescription>{}</NewPortMappingDescription>
-        <NewEnabled>1</NewEnabled>
-        <NewRemoteHost></NewRemoteHost>
-    </u:AddAnyPortMapping>
-</s:Body>
-</s:Envelope>",
-        protocol,
-        external_port,
-        local_addr.ip(),
-        local_addr.port(),
-        lease_duration,
-        description,
-    )
+    let args = schema
+        .iter()
+        .map(|argument| {
+            let value = match argument.as_str() {
+                "NewEnabled" => 1.to_string(),
+                "NewExternalPort" => external_port.to_string(),
+                "NewInternalClient" => local_addr.ip().to_string(),
+                "NewInternalPort" => local_addr.port().to_string(),
+                "NewLeaseDuration" => lease_duration.to_string(),
+                "NewPortMappingDescription" => description.to_string(),
+                "NewProtocol" => protocol.to_string(),
+                "NewRemoteHost" => "".to_string(),
+                _ => panic!("Nope"),
```

Comment:
Can we just ignore the parameter instead of panicking? I don't think a library should panic in these cases. Either we report an error or just ignore it and log a warning.

Unless the parameters are filtered somewhere else where I missed it. The input comes from the router and could make the program panic.

Reply:
Ah, you are totally right, I accidently left this `panic` in the code, I intended to delete it before the PullRequest. A warning to the logger / stderr would be the proper way to go here. In case we missed arguments, we get complaints from the router anyway.

Reply:
We already have the log library in the deps, `warn!` should do it.

Reply:
Yep, working on it right now. ;-)